### PR TITLE
RET-3296: Rule21Referral cronjob

### PR DESCRIPTION
**Pull request by JackReevies**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RET-3296

### Change description ###

Rule21Referral cronjob

**Does this PR introduce a breaking change?** (check one with "x")

```
[] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/wa/BFActionsScheduledTasks.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+@RequiredArgsConstructor
+public class BFActionsScheduledTasks {
+    private final AdminUserService adminUserService;
+    private final CcdClient ccdClient;
+
+    @Scheduled(cron = "${cron.bfActionTask}")
+    public void createTasksForBFDates() {
+        log.info("Checking for expired BFDates");
+
+        String now = UtilHelper.formatCurrentDate2(LocalDate.now().plusDays(-1));
```
Comment (hsjhita) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/wa/BFActionsScheduledTasks.java:39]: would minusDays be better? not asking for it to be changed but just a thought

**Review by lvo700 — APPROVED** [review]

File: build.gradle [hunk 2]
```diff
@@ -335,8 +335,8 @@ dependencyManagement {
 
 dependencies {
 
-    implementation group: 'com.github.hmcts', name: 'et-common', version:'1.2.10'
-    implementation group: 'com.github.hmcts', name: 'et-data-model', version:'3.2.3'
+    implementation group: 'com.github.hmcts', name: 'et-common', version:'c401358cb6'
```
Comment (hsjhita) [inline build.gradle:338]: also might be worth merging these PRs into master to properly version these

File: src/main/resources/application.yaml [hunk 3]
```diff
@@ -196,4 +196,5 @@ launchdarkly:
   offline-mode: false
   env: ${LAUNCH_DARKLY_ENV:default}
 
-
+cron:
```
Comment (hsjhita) [inline src/main/resources/application.yaml:199]: does this need to be in the charts to set it per env?

Comment (JackReevies) [reply src/main/resources/application.yaml:199]: yeah unfortunately, that's how we're going to configure that it only runs on a single instance

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/wa/BFActionsScheduledTasks.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+            }
+        });
+    }
+
+    private String buildQueryForExpiredBFActionsWithNoResponse(String now) {
+        return new SearchSourceBuilder()
+                .size(maxCases)
+                .query(new BoolQueryBuilder()
+                        .must(new TermQueryBuilder("data.respondentCollection.value.responseReceived", NO))
+                        .must(QueryBuilders.rangeQuery("data.bfActions.value.bfDate").to(now).includeUpper(true))
```
Comment (hsjhita) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/wa/BFActionsScheduledTasks.java:73]: just changed 'now' to 'yesterday'
```suggestion
    private String buildQueryForExpiredBFActionsWithNoResponse(String yesterday) {
        return new SearchSourceBuilder()
                .size(maxCases)
                .query(new BoolQueryBuilder()
                        .must(new TermQueryBuilder("data.respondentCollection.value.responseReceived", NO))
                        .must(QueryBuilders.rangeQuery("data.bfActions.value.bfDate").to(yesterday).includeUpper(true))
```

File: src/main/resources/application.yaml (hunk 3 above)
Comment (hsjhita) [reply src/main/resources/application.yaml:199]: is it worth adding these to the chart now?

**Review by fergusoshea — DISMISSED** [review]

**Review by hsjhita — APPROVED** [review]

**Review by jip1984 — DISMISSED** [review]

**Review by jip1984 — APPROVED** [review]

**Review by lvo700 — APPROVED** [review]

**Pull request merged**